holidayextras/brands
# Product list

#### What does this PR do? (please provide any background)
A new component for displaying products has been added to the Globals brand. Read the accompanying Pattern Library page for more information.

In addition to this new component, a variety of functional utility classes and more additions to Globals have been added:
* Responsive margin and padding resets.
* Responsive floats.
* Responsive display.
* Product rating colour-scale variants.
* Text-inverse utility class.
* New variation of thumbnail design with zooming animation on hover/focus.
* 'Wishlist' heart icon added, with active animation.
* Fixed-ratio labels for use with product ratings or icons etc.

#### How can this be tested?
`cd pattern-library && npm link ../brands`
Checkout the corresponding branch in Pattern Library and `npm link` to all brands you wish to test - see [PR](https://github.com/holidayextras/pattern-library/pull/279) for more details.
Go to [Product list](http://localhost:8687/patterns/product-list/) page in Pattern Library.

#### Any tech debt?
* This is a first iteration, which will inevitably need tweaks and refinements as it is stretch-tested.
* There is a bit of work to do at a brand level to ensure all brands display the product list as intended. The majority of the current visible bugs are to do with incorrect colours from bg-variants in brands.

#### Screenshots / Screencast
A few (of many) variations of product lists:
## Divided product list
![image](https://user-images.githubusercontent.com/2961244/27335128-6aec833c-55c3-11e7-9df3-67e8e922ffde.png)
## Product panels grid
![image](https://user-images.githubusercontent.com/2961244/27335178-8e8f8ea6-55c3-11e7-928d-2f441825761a.png)
## Mobile product list
<img src="https://user-images.githubusercontent.com/2961244/27335159-824f329a-55c3-11e7-8d84-283c02770c49.png" width="320px">

#### What gif best describes how you feel about this work?
![](https://media.giphy.com/media/xjTIBA4gjvXIQ/giphy.gif)


---

**Reviewer**

By adding a +1 you are confirming you have...
- Witnessed the work behaving as expected (this could be on the author's machine or screencast).
- Ensured the release has the correct version bump.
- Checked all the work locally on pattern library.
